### ISSUE TITLE
Add action for background box for scale bar overlay

### ISSUE THREAD

**Issue by jni** [napari/napari#6991]
## 🚀 Feature

#4511 added some really cool functionality to the scale bar, including a translucent box for improved visibility when overlaid on certain images. I totally missed it at the time and I was today years old when I discovered that functionality. 😂 A strong reason for this is that there's no corresponding toggle in the View>Scale Bar menu. We should add one! Hopefully it is not too hard to do by modeling it after the existing toggles (visible, colored, and ticks).

**Comment by psobolewskiPhD** [comment]
Yeah! Here's the implementation PR: https://github.com/napari/napari/pull/4511
There's color options too (and units of course).
Would actually be nice to have a widget for setting those. 👀

**Comment by jni** [comment]
> There's color options too (and units of course), so simple toggle action is a start, but would actually be nice to have a widget for setting those. 👀

Yes, I just wanted to keep this one issue small. 😅 Actually it would be good to have some magicgui providers for particular types of data, e.g. color picker for a color, then we could have actions that require color in the menus and they would just pop up a color picker on demand — and all would use the same code. Ditto for simple stuff like e.g. integers/strings (projection axis picker, or stacking axis picker.)

**Comment by Czaki** [comment]
> There's color options too (and units of course), so simple toggle action is a start, but would actually be nice to have a widget for setting those. 👀

Units should be inherited from layer metadata.
